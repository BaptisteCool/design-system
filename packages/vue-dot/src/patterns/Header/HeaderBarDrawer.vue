<template>
	<VNavigationDrawer
		v-if="$vuetify.breakpoint.xs"
		v-model="showActionList"
		v-bind="$attrs"
		width="360"
	>
		<VListItem>
			<VListItemAvatar>
				<VIcon>{{ mdiAccount }}</VIcon>
			</VListItemAvatar>

			<VListItemContent>
				<VListItemTitle>{{ firstname }} {{ lastname }} </VListItemTitle>
				<v-list-item-subtitle>{{ info }} </v-list-item-subtitle>
			</VListItemContent>
		</VListItem>

		<VDivider />

		<VList
			nav
		>
			<VListItem
				v-for="(item, index) in actionsList"
				:key="index"
				@click="$emit('click:action', index); showActionList = false"
			>
				{{ item }}
			</VListItem>
			<VListItem @click="$emit('click:logout'); showActionList = false">
				<VListItemIcon>
					<VIcon color="primary">
						{{ mdiExitToApp }}
					</VIcon>
				</VListItemIcon>
				<VListItemContent>
					{{ locales.logOut }}
				</VListItemContent>
			</VListItem>
			<VListItem @click="showActionList = false">
				<VListItemIcon>
					<VIcon>
						{{ mdiChevronLeft }}
					</VIcon>
				</VListItemIcon>
			</VListItem>
		</VList>
	</VNavigationDrawer>
</template>

<script lang="ts">
	import Vue from 'vue';
	import Component from 'vue-class-component';
	import locales from './locales';
	import { mdiAccount, mdiChevronLeft, mdiExitToApp } from '@mdi/js';

	import HeaderBarMenu from './HeaderBarMenu.vue';

	import { mapState } from 'vuex';

	const Props = Vue.extend({
		props: {
			value: {
				type: Boolean,
				default: false
			},
			firstname: {
				type: String,
				default: null
			},
			lastname: {
				type: String,
				default: null
			},
			info: {
				type: String,
				default: null
			},
			actionsList: {
				type: [Array, Object],
				default: () => []
			}
		}
	});

    @Component<HeaderBarDrawer>({
        watch: {
            value(newValue){
                this.showActionList = newValue;
            },
            showActionList(newValue){
                this.$emit('input', newValue);
            }
        }
    })
	export default class HeaderBarDrawer extends Props {
        locales = locales;
        mdiAccount = mdiAccount;
		mdiExitToApp = mdiExitToApp;
        mdiChevronLeft = mdiChevronLeft;

        showActionList = false;

	}
</script>

<style scoped>

</style>