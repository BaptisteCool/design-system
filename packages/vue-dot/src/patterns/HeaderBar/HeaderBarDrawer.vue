<template>
	<VNavigationDrawer
		:value="value"
		v-bind="{...$attrs, ...options.drawer}"
		@input="toggleDrawer($event)"
	>
		<VListItem>
			<VListItemAvatar>
				<VIcon>{{ mdiAccount }}</VIcon>
			</VListItemAvatar>

			<VListItemContent>
				<VListItemTitle v-bind="options.userName">
					{{ firstname }} {{ lastname }}
				</VListItemTitle>
				<v-list-item-subtitle v-bind="options.userInfo">
					{{ info }}
				</v-list-item-subtitle>
			</VListItemContent>
		</VListItem>

		<VDivider />

		<HeaderBarActions
			:actions-list="actionsList"
			:logged-in="loggedIn"
			@click:action="clickAction($event); toggleDrawer(false)"
			@click:logout="clickLogout(); toggleDrawer(false)"
		/>
	</VNavigationDrawer>
</template>

<script lang="ts">
	import Vue from 'vue';
	import Component from 'vue-class-component';
	import { mdiAccount } from '@mdi/js';

	import customizable, { Options } from '../../mixins/customizable';
	import config from './config/HeaderBardrawer';

	import HeaderBarActions from './HeaderBarActions.vue';
	import header from './mixins/header';

	const Props = Vue.extend({
		props: {
			value: {
				type: Boolean,
				default: false
			}
		}
	});

	@Component<HeaderBarDrawer>({
			mixins: [
				// Default configuration
				customizable(config),
				header
			],
			components: { HeaderBarActions }
	})
	export default class HeaderBarDrawer extends Props {
		// Mixin computed data
		options!: Options;

		// icons
		mdiAccount = mdiAccount;

		// show the drawer or not
		toggleDrawer(value: boolean) {
			this.$emit('input', value);
		}

	}
</script>

<style lang="scss" scoped>
</style>
