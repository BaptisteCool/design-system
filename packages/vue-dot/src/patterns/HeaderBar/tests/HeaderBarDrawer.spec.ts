import Vue from 'vue';
import { Wrapper } from '@vue/test-utils';

import { mountComponent } from '@/tests';
import html from '@/tests/html';

import HeaderBarDrawer from '../HeaderBarDrawer.vue';

let wrapper: Wrapper<Vue>;

// Tests
describe('HeaderBarDrawer test', () => {
	it('renders with all info correctly', () => {
		// Mount component
		wrapper = mountComponent(HeaderBarDrawer, {
			propsData: {
				value: true,
				firstname: 'Prénom',
				lastname: 'nom',
				actionsList: ['text1', 'text2'],
				info: 'information supplémentaire'
			}
		});

		expect(html(wrapper)).toMatchSnapshot();
	});

});
