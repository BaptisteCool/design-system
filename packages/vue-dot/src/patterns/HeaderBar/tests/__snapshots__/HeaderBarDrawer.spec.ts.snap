// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBarDrawer test renders with all info correctly 1`] = `
<vnavigationdrawer-stub fixed="true" height="100%" minivariantwidth="80" mobilebreakpoint="1264" src="" tag="aside" temporary="true" width="360" value="true">
  <vlistitem-stub activeclass="" tag="div">
    <vlistitemavatar-stub size="40">
      <vicon-stub>M12,4A4,4 0 0,1 16,8A4,4 0 0,1 12,12A4,4 0 0,1 8,8A4,4 0 0,1 12,4M12,14C16.42,14 20,15.79 20,18V20H4V18C4,15.79 7.58,14 12,14Z</vicon-stub>
    </vlistitemavatar-stub>
    <vlistitemcontent-stub>
      <vlistitemtitle-stub class="Subtitle 1">
        Prénom nom
      </vlistitemtitle-stub>
      <v-list-item-subtitle-stub class="body-2">
        information supplémentaire
      </v-list-item-subtitle-stub>
    </vlistitemcontent-stub>
  </vlistitem-stub>
  <vdivider-stub></vdivider-stub>
  <headerbaractions-stub accounttext="Mon compte" actionslist="text1,text2" navigationlist=""></headerbaractions-stub>
</vnavigationdrawer-stub>
`;
